lldb/test/Shell/Driver/TestConvenienceVariables.test

```diff
@@ -1,3 +1,4 @@
+# XFAIL: target=x86_64-{{.*}}-windows{{.*}}

```

Comment:
@JDevlieghere please LMK if the target triple can be expressed more clearly as a combination of other filters in `.test` files, here and throughout.

Reply:
something like `system-windows && target-x86_64` ought to do it. However, like I said in the other review, I think we ought to figure out why are these failing for you.

Reply:
Interestingly, this test is green on our Windows native CI. It uses cmd.exe. Here's the log:

```
********************
PASS: lldb-shell :: Driver/TestConvenienceVariables.test (88 of 543)
Exit Code: 0

Command Output (stdout):
--
# RUN: at line 2
mkdir -p C:\build-lldb-native\tools\lldb\test\Shell\Driver\Output\TestConvenienceVariables.test.tmp
# executed command: mkdir -p 'C:\build-lldb-native\tools\lldb\test\Shell\Driver\Output\TestConvenienceVariables.test.tmp'
# note: command had no output on stdout or stderr
# RUN: at line 3
'C:\Python312\python.exe' C:\lldb\test\Shell\helper\build.py --compiler=any --arch=64 --tools-dir=C:/build-lldb-native/./bin --libs-dir=C:/build-lldb-native/./lib C:\lldb\test\Shell\Driver/Inputs/hello.cpp -o C:\build-lldb-native\tools\lldb\test\Shell\Driver\Output\TestConvenienceVariables.test.tmp/target.out
# executed command: 'C:\Python312\python.exe' 'C:\lldb\test\Shell\helper\build.py' --compiler=any --arch=64 --tools-dir=C:/build-lldb-native/./bin --libs-dir=C:/build-lldb-native/./lib 'C:\lldb\test\Shell\Driver/Inputs/hello.cpp' -o 'C:\build-lldb-native\tools\lldb\test\Shell\Driver\Output\TestConvenienceVariables.test.tmp/target.out'
# .---command stdout------------
# | Cleaning hello.ilk
# | Cleaning target.out-hello.obj
# | Cleaning target.pdb
# | Cleaning target.out
# | 
# | 
# | 
# | compiling hello.cpp -> target.out-hello.obj
# |   STDOUT:
# | 
# | 
# | 
# | 
# | linking target.out-hello.obj -> target.out
# |   STDOUT:
# | 
# `-----------------------------
# RUN: at line 4
c:\build-lldb-native\bin\lldb.exe --no-lldbinit -S C:/build-lldb-native/tools/lldb\test\Shell\lit-lldb-init-quiet C:\build-lldb-native\tools\lldb\test\Shell\Driver\Output\TestConvenienceVariables.test.tmp/target.out -s C:\lldb\test\Shell\Driver/Inputs/convenience.in -o quit | c:\build-lldb-native\bin\filecheck.exe C:\lldb\test\Shell\Driver\TestConvenienceVariables.test
# executed command: 'c:\build-lldb-native\bin\lldb.exe' --no-lldbinit -S 'C:/build-lldb-native/tools/lldb\test\Shell\lit-lldb-init-quiet' 'C:\build-lldb-native\tools\lldb\test\Shell\Driver\Output\TestConvenienceVariables.test.tmp/target.out' -s 'C:\lldb\test\Shell\Driver/Inputs/convenience.in' -o quit
# note: command had no output on stdout or stderr
# executed command: 'c:\build-lldb-native\bin\filecheck.exe' 'C:\lldb\test\Shell\Driver\TestConvenienceVariables.test'
# note: command had no output on stdout or stderr

--

```

However, I'm unsure if the quotes are printed in log the same way they are passed to cmd.

Reply:
@DavidSpickett would you be able to share the CMake configuration your native Windows native CI is using? I'm having trouble spotting what's wrong with my local build.

Reply:
For reference, my current setup steps are the following commands run in a non-admin cmd.exe

```
C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\Tools\VsDevCmd.bat -host_arch=amd64 -arch=amd64
C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\Tools\VsDevCmd.bat -test

set LLDB_USE_LLDB_SERVER=1
set MSVC_INSTALL_DIR=C:\Program Files\Microsoft Visual Studio\2022\Community
set "PATH=%MSVC_INSTALL_DIR%\DIA SDK\bin\amd64;%PATH%"
set "PATH=%MSVC_INSTALL_DIR%\VC\Tools\Llvm\x64\bin;%PATH%"

cmake -G Ninja ^
      -S llvm ^
      -B build ^
      -D Python3_EXECUTABLE="C:\Program Files\Python312\python.exe" ^
      -D CMAKE_C_COMPILER="C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\Llvm\x64\bin\clang-cl.exe" ^
      -D CMAKE_CXX_COMPILER="C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\Llvm\x64\bin\clang-cl.exe" ^
      -D CMAKE_C_COMPILER_LAUNCHER=ccache ^
      -D CMAKE_CXX_COMPILER_LAUNCHER=ccache ^
      -D CMAKE_BUILD_TYPE="Release" ^
      -D CMAKE_LINKER="C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\Llvm\x64\bin\lld-link.exe" ^
      -D LLVM_ENABLE_PROJECTS="clang;lldb;lld" ^
      -D LLVM_HOST_TRIPLE="x86_64-unknown-windows-msvc" ^
      -D LLVM_ENABLE_LLD="YES" ^
      -D LLVM_ENABLE_LIBEDIT="NO" ^
      -D LLVM_ENABLE_DIA_SDK="OFF" ^
      -D LLDB_ENABLE_PYTHON=1 ^
      -D LLDB_PYTHON_EXE_RELATIVE_PATH="python.exe" ^
      -D LLDB_TEST_USER_ARGS=--skip-category=watchpoint ^
      -D LLDB_ENFORCE_STRICT_TEST_REQUIREMENTS=1 ^
      -D CMAKE_CXX_STANDARD=14

ninja -C build check-lldb
```

Reply:
I've removed CMAKE_CXX_STANDARD=14 above as it is leftover from debugging and not correct or necessary.

Reply:
> @dzhidzhoev would you be able to share the CMake configuration your Windows native CI is using? I'm having trouble spotting what's wrong with my local build.

For sure! Sorry, I needed some time to sort it out, because I hadn't reproduced it on my local machine before, only had it in CI.

```
cmake -DLLVM_LIT_ARGS="-v -vv --threads=8 --xunit-xml-output=xunit-report.xml" ^
  -G Ninja -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache ^
  -DCMAKE_INSTALL_PREFIX=native -DCMAKE_BUILD_TYPE=Release ^
  -DLLVM_ENABLE_PROJECTS=lldb;llvm;clang;lld -DLLVM_ENABLE_ASSERTIONS=ON ^
  -DMake_EXECUTABLE=c:/bin/make-wfix.exe ^
  -DLLDB_TEST_USER_ARGS=--skip-category=watchpoint ^
  -DLLDB_ENFORCE_STRICT_TEST_REQUIREMENTS=ON ^
 c:/lldb-test-scripts/llvm
```

Reply:
`LLVM_ENABLE_DIA_SDK="OFF"` Seems to have been the culprit for me. After removing this and using the default `"ON"`  all of the Shell tests I was previously seeing fail are now passing for me. Thanks so much for your help in debugging!